core/unix/injector.c

```diff
@@ -213,16 +213,18 @@ report_dynamorio_problem(dcontext_t *dcontext, uint dumpcore_flag, app_pc except
 static void
 pre_execve_ld_preload(const char *dr_path)
 {
-    char ld_preload_libs[MAX_OPTIONS_STRING];
     char ld_lib_path[MAX_OPTIONS_STRING];
     const char *last_slash = NULL;
     const char *mode_slash = NULL;
     const char *lib_slash = NULL;
+    const char *cur_path = getenv("LD_LIBRARY_PATH");
+    const char *cur = dr_path;
+#ifdef LINUX
```

Comment:
Hmm, so Mac is still overwriting instead of appending, so we have split behavior now.  Maybe worth addressing -- or maybe just a XXX comment.  I think it will let me edit this with a comment.

Reply:
Good catch. I dev on a Mac, so I can fix this there as well if you'd like.

Reply:
That would be awesome, thanks.